File: src/server/controllers/get-config.ts
import express from 'express';

import { wowDataClient } from '@/lib/wow-data-client/wow-data-client';

import { isDev, isSharedHosting } from '../config';
import { getCeConfig } from './export-character';

export function ControllerGetConfig(router: express.Router) {
  router.get('/get-config', (_req, res) => {
    res.setHeader('Cache-Control', 'private, no-cache');
    res.json({
      exportAssetDir: getCeConfig().exportAssetDir,
      isSharedHosting,
      isDev,
      isClassic: wowDataClient.isClassic(),
      buildKey: wowDataClient.cascInfo?.buildKey ?? '',
    });
  });
}

File: golang/internal/server/api/get_config.go
package api

import "net/http"

func registerGetConfig(r Router, d *Deps) {
	r.Get("/get-config", func(w http.ResponseWriter, req *http.Request) {
		sendJSON(w, http.StatusOK, map[string]any{
			"exportAssetDir": d.ExportAssetDir(),
			"isSharedHosting": d.Config.IsSharedHosting,
			"isDev":           d.Config.IsDev,
			"isClassic":       d.IsClassic(req.Context()),
		})
	})
}
